low-level/files/opt/seagate/sspl/setup/resource_map/lr2/resource_map.py

```diff
@@ -95,3 +82,54 @@ def set_health_data(health_data: dict, status, description=None,
             "recommendation": recommendation,
             "specifics": specifics
         })
+
+
+class CortxManifest(metaclass=ABCMeta):
+    """Abstract class for all other manifest types."""
+
+    name = "manifest"
+
+    def __init__(self):
+        """Initialize manifest."""
+        logging_level = Conf.get(SSPL_CONF,
+            f"{SYSTEM_INFORMATION}>{LOG_LEVEL}", "INFO")
+        init_logging('node-manifest', logging_level)
+
+    @abstractmethod
+    def get_manifest_info(self, rpath):
+        """
+        Get technical information of fru in given manifest id.
+
+        rpath: manifest id (Example: compute>system)
+        """
+        pass
+
+def get_realstor_encl_data(fru: str):
```

Comment:
Go by object oriented way. Add this under resource_map class.

Reply:
Done.